data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/stream/StreamScheduler.java

```diff
@@ -172,23 +139,20 @@ public void run() {
 
     private BiConsumer completeConsumer(StreamPartition streamPartition) {
         return (v, ex) -> {
-            if (!dynamoDBSourceConfig.isAcknowledgmentsEnabled()) {
-                numOfWorkers.decrementAndGet();
-                if (numOfWorkers.get() == 0) {
-                    activeChangeEventConsumers.decrementAndGet();
-                }
-                shardsInProcessing.decrementAndGet();
+            numOfWorkers.decrementAndGet();
+            if (numOfWorkers.get() == 0) {
+                activeChangeEventConsumers.decrementAndGet();
             }
+            shardsInProcessing.decrementAndGet();
             if (ex == null) {
                 LOG.info("Shard consumer for {} is completed", streamPartition.getShardId());
-                coordinator.completePartition(streamPartition);
+                if (!dynamoDBSourceConfig.isAcknowledgmentsEnabled()) {
+                    coordinator.completePartition(streamPartition);
+                }
             } else {
-                // Do nothing
-                // The consumer must have already done one last checkpointing.
                 LOG.error("Received an exception while processing shard {}, giving up shard: {}", streamPartition.getShardId(), ex);
                 coordinator.giveUpPartition(streamPartition);
```

Comment:
Think we may need to signal to acknowledgment set manager to stop tracking this partition if we get an error. Otherwise it'll probably hit an error trying to update the state, and eventually will stop tracking from ack timeout expiring.